lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -1598,9 +1598,19 @@ protected function _validateAndCompleteOneToOneMapping(array $mapping)
                 if ( ! $this->table) {
                     throw new RuntimeException("ClassMetadataInfo::setTable() has to be called before defining a one to one relationship.");
                 }
+                /*
+                 * Mapping via the field name causes issues if unrelated tables use the same fieldName in their independent mappings
+                 * Note: The hack below of prepending the table name probably exceeds length constraints of some databases
+                 * e.g. Oracle.
+                 * Naming should really be delegated to the NamingStrategy but there is no API in the interface to support this yet.
+
                 $this->table['uniqueConstraints'][$mapping['fieldName']."_uniq"] = array(
                     'columns' => $uniqueConstraintColumns
                 );
+                 */
+                $this->table['uniqueConstraints'][$this->table['name'].$mapping['fieldName']."_uniq"] = array(
```

Comment:
I must say that I agree with prepending the table name here: makes it much more solid.

The problem is that this is a BC break that will cause a lot of problematic schema updates for our users :-\

Reply:
Yes its a real bug.